specs/trust-store-trust-policy.md

```diff
@@ -404,35 +514,45 @@ Notary Project allows user to execute custom validations during verification usi
-            1. If all the certificates in the named store have been evaluated without a match, then fail this step.
-        1. If `trustedIdentities` is `*`, any signing certificate issued by a CA in `trustStore` is allowed, skip to the next validation (Validate Expiry).
-        1. Else validate if the X.509 subject (`x509.subject`) in the `trustedIdentities` list matches with the value of corresponding attributes in the signing certificate’s subject, refer [this section](#trusted-identities-constraints) for details. If a match is not found, fail this step.
+        1. Validate that the signature envelope contains a complete certificate chain that starts from a code signing certificate and terminates with a root certificate. Also, validate that certificates in the chain satisfy [certificate requirements](./signature-specification.md#certificate-requirements).
+        1. For the `trustStore` configured in applicable trust policy perform the following steps.
+            1. Determine the `trust-store-type` based on the [`signing scheme`](./signing-scheme.md/#Supported-Signing-Scheme). For this step, it MUST be `ca` or `signingAuthority`. If none of them is present, fail this step.
+            1. Validate that certificate and certificate-chain lead to a trusted certificate present in the named stores listed under the same `trust-store-type`.
+            1. If all the certificates in the named stores under the same `trust-store-type` have been evaluated without a match, then fail this step.
+        1. If `array-of-trusted-identities` listed under the same `trust-store-type` in `trustedIdentities` is `["*"]`, then skip to the next validation (Validate Expiry).
+        1. Else validate if X.509 subjects (`x509.subject`) listed under the same `trust-store-type` in `trustedIdentities` matches with the value of corresponding attributes in the signing certificate’s subject, refer [this section](#trusted-identities-constraints) for details. If a match is not found, fail this step.
 1. **Validate Expiry:**
     1. If an `expiry time` signed attribute is present in the signature envelope, check if the local machine’s current time(in UTC) is greater than `expiry time`. If yes, fail this step.
-1. **Validate Trusted Timestamp:**
-    1. Check for the timestamp signature in the signature envelope.
-        1. If the timestamp exists, continue with the next step.
-        Otherwise, store the local machine's current time(in  UTC) in variables `timeStampLowerLimit` and `timeStampUpperLimit` and continue with step 6.2.
+1. **Validate Authentic Timestamp:**
+
+    If under signing scheme [`notary.x509`](./signing-scheme.md/#notaryx509):
+    1. Validate that the local machine's current time (in  UTC) is within the signing certificate chain's validity period. If the validation passes, continue to the next validation (Validate Revocation Status). Otherwise, continue to step 6.2.
```

Comment:
If we use timestamp instead of datetime, we dont need to worry about time zones.

Reply:
As explained above, in the context of `Notary Project`, `timestamp` and `time point` should be two different concepts for easier understanding.